hub/elastic_sync/db.py

```diff
@@ -27,197 +27,197 @@ def estimate_timestamp(self, height: int) -> int:
-                ).outputs[reposted_claim.position].metadata
-            except:
-                self.logger.error("failed to parse reposted claim in tx %s that was reposted by %s",
-                                  claim.reposted_claim_hash.hex(), claim_hash.hex())
-                return
-        if reposted_metadata:
-            if reposted_metadata.is_stream:
-                meta = reposted_metadata.stream
-            elif reposted_metadata.is_channel:
-                meta = reposted_metadata.channel
-            elif reposted_metadata.is_collection:
-                meta = reposted_metadata.collection
-            elif reposted_metadata.is_repost:
-                meta = reposted_metadata.repost
-            else:
-                return
-            reposted_tags = [tag for tag in meta.tags]
-            reposted_languages = [lang.language or 'none' for lang in meta.languages] or ['none']
-            reposted_has_source = False if not reposted_metadata.is_stream else reposted_metadata.stream.has_source
-            reposted_claim_type = CLAIM_TYPES[reposted_metadata.claim_type]
-            reposted_stream_type = STREAM_TYPES[guess_stream_type(reposted_metadata.stream.source.media_type)] \
-                if reposted_has_source else 0
-            reposted_media_type = reposted_metadata.stream.source.media_type if reposted_metadata.is_stream else 0
-            if not reposted_metadata.is_stream or not reposted_metadata.stream.has_fee:
-                reposted_fee_amount = 0
+    async def prepare_claim_metadata_batch(self, claims: Dict[bytes, ResolveResult], extras):
+        metadatas = {}
+        needed_txos = set()
+
+        for claim_hash, claim in claims.items():
+            reposted_claim_hash = claim.reposted_claim_hash
+            needed_txos.add((claim.tx_hash, claim.position))
+            if reposted_claim_hash:
+                if not reposted_claim_hash not in extras:
+                    continue
+                reposted_claim = extras.get((reposted_claim_hash))
+                if reposted_claim:
+                    needed_txos.add((reposted_claim.tx_hash, reposted_claim.position))
+        metadatas.update(await self.get_claim_metadatas(list(needed_txos)))
+
+        for claim_hash, claim in claims.items():
+            metadata = metadatas.get((claim.tx_hash, claim.position))
+            if not metadata:
+                continue
+            if not metadata.is_stream or not metadata.stream.has_fee:
+                fee_amount = 0
             else:
-                reposted_fee_amount = int(max(reposted_metadata.stream.fee.amount or 0, 0) * 1000)
-                if reposted_fee_amount >= 9223372036854775807:
-                    return
-            reposted_fee_currency = None if not reposted_metadata.is_stream else reposted_metadata.stream.fee.currency
+                fee_amount = int(max(metadata.stream.fee.amount or 0, 0) * 1000)
+                if fee_amount >= 9223372036854775807:
```

Comment:
probably not in this PR, but would be nice to extract those huge numbers to something with meaningful names